### PR TITLE
Add util types

### PR DESCRIPTION
**Pull request by devanshutanna**
## Overview

- Description: Add types and test for the old and new types
- Schema update type: extend
- Services affected: none

## How was this tested?

Yes, have created type tests..

### CODE REVIEW

**Review by KillDozerX2 — CHANGES_REQUESTED** [review]
Please make changes as requested in the call.

File: package.json [hunk 1]
```diff
@@ -17,6 +17,9 @@
   "files": [
     "lib/**/*"
   ],
+  "engines": {
+    "node": ">=12"
+  },
```
Comment (KillDozerX2) [inline package.json]: @devanshutanna Let's bump this to >= 16.

File: package.json [hunk 2]
```diff
@@ -70,7 +73,7 @@
     "serverless": "^3.26.0",
     "serverless-step-functions": "^3.12.1",
     "ts-node": "^10.9.1",
-    "typescript": "^4.9.0-dev.20220916"
+    "typescript": "^4.9.5"
```
Comment (KillDozerX2) [inline package.json:76]: @devanshutanna Is this for `satisfies`?

**Review by KillDozerX2 — CHANGES_REQUESTED** [review]
Good to go, just waiting for docs and a few clarifications.
Also, you might have to rebase.

File: package.json (hunk 1 above)
Comment (devanshutanna) [reply package.json]: Ok 👍

File: package.json (hunk 2 above)
Comment (devanshutanna) [reply package.json:76]: We did have support for satisfies already..
Just that we were using a dev build,
So updated it to the latest version.

Comment (KillDozerX2) [reply package.json:76]: Understood, thanks

**Review by KillDozerX2 — APPROVED** [review]

**Pull request merged**